memory/transparent_hugepages.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+# Test Inspired basically from
+# https://github.com/autotest/tp-qemu/blob/master/generic/tests/trans_hugepage.py
+#
+# Copyright: 2017 IBM
+# Author: Santhosh G <santhog4@linux.vnet.ibm.com>
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import memory
+from avocado.core import data_dir
+from avocado.utils.partition import Partition
+
+
+class Thp(Test):
+
+    '''
+    The test enables THP and stress the system using dd load
+    and verifies whether THP has been allocated for usage or not
+    '''
+
+    def setUp(self):
+
+        '''
+        Sets all the reqd parameter and also
+        mounts the device to be used in test.
+        '''
+
+        # Set params as per available memory in system
+        self.tmpdir = data_dir.get_tmp_dir()
+        self.mem_path = os.path.join(self.tmpdir, 'thp_space')
+        free_mem = int(memory.freememtotal() / 1024)
+        self.dd_timeout = 900
+
+        # Set block size as hugepage size * 2
+        self.block_size = (memory.get_huge_page_size() / 1024) * 2
+        self.count = free_mem / self.block_size
+
+        # Mount device as per free memory size
+        os.mkdir(self.mem_path)
+        self.device = Partition(device="none", mountpoint=self.mem_path)
+        self.device.mount(mountpoint=self.mem_path, fstype="tmpfs",
+                          args='-o size=%dM' % free_mem)
+
+    def test(self):
+
+        '''
+        Enables THP , Runs the dd workload and checks whether THP
+        has been allocated.
+        '''
+
+        # Enables THP
+        try:
+            memory.set_thp_value("enabled", "always")
+        except Exception as details:
+            self.fail("Failed  %s" % details)
+
+        # Read thp values before stressing the system
+        thp_alloted_before = int(memory.read_from_vmstat("thp_fault_alloc"))
+        thp_split_before = int(memory.read_from_vmstat("thp_split_page"))
+        thp_collapse_alloc_before = int(memory.read_from_vmstat
+                                        ("thp_collapse_alloc"))
+
+        # Start Stresssing the  System
+        self.log.info('Stress test Start')
+
+        for iterator in range(self.count):
+            stress_cmd = 'dd if=/dev/zero of=%s/%d bs=%dM count=1'\
+                         % (self.mem_path, iterator, self.block_size)
+            process.system(stress_cmd, timeout=self.dd_timeout,
+                           verbose=False, shell=True)
```

Comment:
please use `ignore_status=True` as a param in` process.system` like if command error out script also error out ?